### PR TITLE
Create issue bug report and enhancement templates

### PR DESCRIPTION
**Pull request by jdmcbr**
As a followup to #899, it would be nice to highlight this function, as
well as other useful information, when users are creating new bug
reports or requests for new features. These are copied and very lightly
edited from the templates used by pandas.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+```
+
+#### Problem description
+
+[this should explain **why** the current behaviour is a problem and why the expected output is a better solution]
+
+#### Expected Output
+
+<details>
```
[the remark this answers is not in the document]
Comment (jdmcbr) [reply .github/ISSUE_TEMPLATE/bug_report.md:33]: Oh, doh, copy/paste error, thanks.

File: .github/ISSUE_TEMPLATE/installation_issue.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+---
+
+name: Installation Issue 
+about: Ask about installing geopandas 
+title: ""
```
Comment (jdmcbr) [inline .github/ISSUE_TEMPLATE/installation_issue.md:5]: Do we want a suggested title here with `INST` or the like?

File: .github/ISSUE_TEMPLATE/installation_issue.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+labels: "installation"
```
Comment (jdmcbr) [inline .github/ISSUE_TEMPLATE/installation_issue.md:6]: We don't currently have this label, but I can add this if we're explicitly supporting installation issue questions via the issue tracker.

File: .github/ISSUE_TEMPLATE/installation_issue.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+[what operating system do you have and what package management system are you
+using]
+
+#### Environment details
+
+<details>
+
+[if using conda, paste the output of `conda info` and `conda list`; if using
+pip, `pip freeze`]
```
Comment (jdmcbr) [inline .github/ISSUE_TEMPLATE/installation_issue.md:24]: In the case of pip, I'm not actually sure how likely `pip freeze` is to be useful, I just included it as the closest analog for `conda list`.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
+---
+
+name: Bug Report
+about: Create a bug report to help us improve geopandas
+title: "BUG:"
+labels: "bug"
```
Comment (martinfleis) [inline .github/ISSUE_TEMPLATE/bug_report.md:6]: I liked the initial version with `needs triage`, which indicates that reported bug has not been confirmed by the team (just requires adding the label).

File: .github/ISSUE_TEMPLATE/installation_issue.md (hunk 2 above)
Comment (martinfleis) [reply .github/ISSUE_TEMPLATE/installation_issue.md:5]: I think that label will be enough. These will be mostly temporary anyway.

File: .github/ISSUE_TEMPLATE/installation_issue.md (hunk 3 above)
Comment (martinfleis) [reply .github/ISSUE_TEMPLATE/installation_issue.md:6]: That may even be good to add to the template itself. `I have looked through [installation] issues in the repo.`.

File: .github/ISSUE_TEMPLATE/submit_question.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
+---
+
+name: Submit Question
+about: Ask a general question about geopandas
+title: "QST:"
+labels: "question"
+
+---
+
+- [ ] I have searched the [[geopandas] tag](https://stackoverflow.com/questions/tagged/geopandas) on StackOverflow for similar questions.
```
Comment (martinfleis) [inline .github/ISSUE_TEMPLATE/submit_question.md:10]:
```suggestion
- [ ] I have searched the [geopandas] tag on [StackOverflow](https://stackoverflow.com/questions/tagged/geopandas) an [GIS StackExchange](https://gis.stackexchange.com/questions/tagged/geopandas) for similar questions.
```

File: .github/ISSUE_TEMPLATE/submit_question.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+- [ ] I have asked my usage related question on [StackOverflow](https://stackoverflow.com).
```
Comment (martinfleis) [inline .github/ISSUE_TEMPLATE/submit_question.md:12]:
```suggestion
- [ ] I have asked my usage related question on [StackOverflow](https://stackoverflow.com) or [GIS StackExhange](https://gis.stackexchange.com).
```

**Review by martinfleis — COMMENTED** [review]
I already feel that this will help us a lot. Thanks for working on it.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 5 above)
Comment (jdmcbr) [reply .github/ISSUE_TEMPLATE/bug_report.md:6]: Okay, cool. We don't actually have that label in geopandas (which is why I removed it), but I'll add it back here and create that label.

**Review by martinfleis — APPROVED** [review]
Looks good to me, thanks!

**Comment by martinfleis** [comment]
@jorisvandenbossche @jdmcbr shall we merge?

**Comment by jdmcbr** [comment]
I didn't have anything else in mind to add, but happy to wait for @jorisvandenbossche to take another look.

**Comment by martinfleis** [comment]
I am merging this now to have them in place. We can revise later once we'll see how they work in practice.

Thanks @jdmcbr, I think this will be super helpful.

**Pull request merged**